src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -1112,17 +1112,14 @@ public String run(final Matcher m) {
 					final Condition cond = Condition.parse(name, "can't understand this condition: '" + name + "'");
 					if (cond == null)
 						continue;
-					if (!(cond instanceof SkippableCondition)) {
-						Skript.error("Condition can not be used with 'parse if': " + name);
-						continue;
-					}
 					try {
 						@SuppressWarnings("null")
 						boolean check = cond.check(null);
-						if (!check)
-							items.add(new Conditional(cond));
-						else
-							items.add(new Conditional(cond, (SectionNode) n));
+						@SuppressWarnings("null")
+						Conditional conditional = new Conditional(cond, null);
+						if (check)
+							conditional = new Conditional(cond, (SectionNode) n);
+						items.add(conditional);
 					} catch (NullPointerException ex) {
 						Skript.error("Could not parse: " + cond);

```

Comment:
instead of saying could not parse, why not say something like `'(cond)' cannot be used with the 'parse if' condition`

Reply:
That makes sense.
with `'(cond)'` are you suggesting putting the error'd code before the message?
so they would see:
```
'parse if player has a diamond' cannot be used with the 'parse if' condition
```
?

Or should the "code" stay after our error message?

Reply:
yeah like your example, perhaps cutting off the "parse if" part

Reply:
Message has been updated :)